content/courses/Ada_For_The_C_Embedded_Developer/chapters/02_Perspective.rst

```diff
@@ -36,13 +36,67 @@ to GNAT, and sometimes to specific platforms supported by GNAT. We'll try to
 make the distinction between what is GNAT-specific and Ada generic as much as
 possible through this book.
 
+For an introduction to the GNAT Toolchain for the GNAT Community edition, you
+may refer to the
+:doc:`Introduction to GNAT Toolchain <courses/GNAT_Toolchain_Intro/index>`
+course.
+
 The GNAT Toolchain for Embedded Targets
 -----------------------------------------
 
-.. todo::
+When we're discussing embedded programming, our target device is often
+different from the host, which is the device we're using to actually write and
+build an application. In this case, we're talking about compilation cross
+platforms.
+
+The GNAT toolchain supports cross platform compilation for various
+target devices. This section provides a short introduction to the topic. For
+more details, please refer to the
+`GNAT User’s Guide Supplement for Cross Platforms <https://docs.adacore.com/gnat_ugx-docs/html/gnat_ugx/gnat_ugx.html>`_
+
+GNAT supports two types of cross platforms:
+
+- **cross targets**, where the target device has an embedded operating system.
+
+    - ARM-Linux, which is commonly found in a Raspberry-Pi, is a prominent
+      example.
+
+- **bareboard targets**, where the run-times do not depend on an operating
+  system.
+
+    - In this case, the application communicates directly with the device's
+      processor.
+
+For each platform, a set of run-time libraries is available. Run-time libraries
+implement a subset of the Ada language for different use cases, and they're
+different for each target platform. They may be selected via an attribute in
+the project's GPR project file or as a command-line switch to
+:program:`GPRbuild`.
+
+Run-time libraries consists of:
+
+#. Files that are dependent on the target board.
+
+    - These files are responsible for configuring and interacting with the
+      hardware.
+
+    - They are known as Board Support Package |mdash| commonly refer as *BSP*.
+
+#. Code that is target-independent.
+
+    - This code implements language-defined functionality.
+
+The bareboard run-time libraries are provided as customized run-times that are
+configured to target a very specific micro-controller or processor. Therefore,
+for different micro-controllers and processors, the run-time libraries need to
+be ported to your target.
```

Comment:
You can mention some examples here about what needs to be ported: Startup scripts, memory layouts, clock initializations, register descriptions, etc.